### PR TITLE
Expose API route to get Project Access Token for SAB

### PR DESCRIPTION
**Pull request by FyreByrd**
Exposes `/api/project/[id=idNumber]/token` as a route that SAB can POST to to get a Project Access Token for downloads and uploads.

### CODE REVIEW

File: source/SIL.AppBuilder.Portal/src/routes/(unauthenticated)/api/projects/[id=idNumber]/token/+server.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    select: { Id: true }
+  });
+
+  await DatabaseWrites.productTransitions.createMany({
+    data: products.map((p) => ({
+      ProductId: p.Id,
+      AllowedUserNames: user[0].Name,
+      TransitionType: ProductTransitionType.ProjectAccess,
+      InitialState: 'Project ' + use,
+      WorkflowUserId: user[0].WorkflowUserId,
```
Comment (chrisvire) [inline source/SIL.AppBuilder.Portal/src/routes/(unauthenticated)/api/projects/[id=idNumber]/token/+server.ts:216]: Populate with user.Id as a string. Don't assign AllowedUserName. Display name base on WorkflowUserId.  If it is a GUID, look up user.WorkflowUserId. If it is just an int, lookup user.Id

Comment (chrisvire) [reply source/SIL.AppBuilder.Portal/src/routes/(unauthenticated)/api/projects/[id=idNumber]/token/+server.ts:216]: Alternatively, add ProductTransitions.UserId and S2 will populate that.

**Review by chrisvire — APPROVED** [review]
Change WorkflowUserId to UserId in another PR

**Comment by FyreByrd** [comment]
Fixed merge conflicts

**Pull request merged**